gene2phenotype_project/gene2phenotype_app/serializers/publication.py

```diff
@@ -0,0 +1,364 @@
+from rest_framework import serializers
+from datetime import datetime
+
+from ..models import (Publication, PublicationComment,
+                      PublicationFamilies, Attrib, LGDPublication)
+from ..utils import (get_publication, get_authors)
+
+
+class PublicationCommentSerializer(serializers.ModelSerializer):
+    """
+        Serializer for the PublicationComment model.
+        To insert a comment the user is fetched from the context.
+    """
+
+    comment = serializers.CharField()
+    date = serializers.DateTimeField()
+    user = serializers.CharField(source="user.username")
+
+    def create(self, data, publication):
+        """
+            Method to add a comment to a publication.
+            The comment can be public or private.
+
+            Returns:
+                    PublicationComment object
+        """
+
+        comment_text = data.get("comment")
+        is_public = data.get("is_public")
+        user_obj = self.context['user'] # gets the user from the context
+
+        # Check if comment is already stored. We consider same comment if they have the same:
+        #   publication, comment text, user and it's not deleted TODO
+        # Filter can return multiple values - this can happen if we have duplicated entries
+        publication_comment_list = PublicationComment.objects.filter(comment = comment_text,
```

Comment:
I did that to avoid users to interfere with each other's comments.
It's very unlikely they will try to insert the exact same comment, but I'll remove the user from the filter.